packages/suite-desktop/src-electron/hang-detect.ts

```diff
@@ -33,10 +34,16 @@ export const hangDetect = (mainWindow: BrowserWindow): Promise<HandshakeResult>
         timeout = setTimeout(timeoutCallback, HANG_WAIT);
         ipcMain.handleOnce('handshake/client', () => {
             clearTimeout(timeout);
+            const torSettings = store.getTorSettings();
+            const initialHandShakeClient = {
+                torSettings,
+            };
+
             // always resolve repeated handshakes from renderer (e.g. Ctrl+R)
-            ipcMain.handle('handshake/client', () => Promise.resolve());
+            ipcMain.handle('handshake/client', () => Promise.resolve(initialHandShakeClient));
```

Comment:
I thought that you don't want to return the `torSettings` for repeated handshakes. Currently, when I press Ctrl+R, I see Tor loader for the fraction of second (because Tor was still running). And I also suspect that even when you turn the Tor off, the repeated handshakes would return the first `torSettings` again and again, as it was kept in the function's closure.

Reply:
Yes, you are right, what about this 5dbf597cdfffc9799691a0b2c0b403987e5a9b4c  ??